ProgrammingGuide/Compiling.md

```diff
@@ -338,6 +338,7 @@ The following options control `find_package` paths for CMake-based TPLs:
 
 #### Architecture Keywords
 Architecture-specific optimizations can be enabled by specifying `-DKokkos_ARCH_X`.
+See the [complete list of architecture keywords](../API-Reference/Macros.md#architectures).
 
 * Kokkos_ARCH_A64FX
```

Comment:
Should we consider removing the list below (as duplicated) and only leave the link to `API-Reference/Macros.md`?

Reply:
The only concern I have about removing the list below is that there is a very subtle capitalizations differences between the keywords listed on this page and the ones in `API-Reference/Macros.md`.
Here, it is `Kokkos_ARCH_X` whereas in `Macros.md`, it's `KOKKOS_ARCH_X`.
Though for CMake builds, using either doesn't seem to make any difference...

Reply:
I think it's debatable whether users should be able to rely on the existence of these macros.

The `CMake` option always looks like `Kokkos_ENABLE...` (although we also define local CMake variables `KOKKOS_ENABLE...`) while the macro is `KOKKOS_ENABLE...`.

Reply:
note I don't think that KOKKOS_ENABLE is valid from our side.

Reply:
@crtrott , @masterleinad -- Just to make sure I understand the principle here, you are proposing:
1) Users *should not* use Kokkos macros, that macros are/ should be mostly / exclusively for developers;
2) the CMake variables, `Kokkos_ENABLE_X` , `Kokkos_ARCH_XYZ` should be the only user-facing variables (vs. Macros  + CMake variables)?
3) CMake variables will take precedence, and if not set, the macro or environment variable will be used?